include/wabt/shared-validator.h

```diff
@@ -104,12 +104,36 @@ class SharedValidator {
   Result OnLocalDecl(const Location&, Index count, Type type);
 
   Result OnAtomicFence(const Location&, uint32_t consistency_model);
-  Result OnAtomicLoad(const Location&, Opcode, Var memidx, Address align);
-  Result OnAtomicNotify(const Location&, Opcode, Var memidx, Address align);
-  Result OnAtomicRmwCmpxchg(const Location&, Opcode, Var memidx, Address align);
-  Result OnAtomicRmw(const Location&, Opcode, Var memidx, Address align);
-  Result OnAtomicStore(const Location&, Opcode, Var memidx, Address align);
-  Result OnAtomicWait(const Location&, Opcode, Var memidx, Address align);
+  Result OnAtomicLoad(const Location&,
+                      Opcode,
+                      Var memidx,
+                      Address align,
+                      Address offset);
```

Comment:
I wonder if would make sense to bundle `memidx`, `align`, `offset` into some kind of `MemLocation` struct?  Are there other places in the code that want to reason about these three as a unit?

Reply:
I think it would make a lot of sense, but it would be a pretty intensive refactor.

There's a lot of callsites in the BinaryReader (and its delegates, e.g. BinaryReaderIR/Interp/etc.) that pass these three in the form where `memidx` is an `Index` and `align` is in the binary (log2) form.

And there's a lot of callsites in the SharedValidator and in the IR expr constructors that pass these three in the form where memidx is a `Var` and `align` is in the expanded form.

So I think the most elegant/comprehensive treatment might have some sort of (name TBD) MemLocationRaw, used in all the binary readers, and a (name also TBD) MemLocationCooked that's used in the SharedValidator and IR?

Happy to do this now if you want but it might make sense for a separate PR.